File: csharp/05-zombie-code/src/campaign_sender.cs
public record CampaignCustomer(bool Active, string Region, bool GdprOptIn, bool Unsubscribed = false, bool Vip = false);

public class CampaignResult
{
    public int Sent;
    public int Skipped;
    public string Message = "";
    public long Timestamp;
}

public class CampaignSender
{
    public CampaignResult send_campaign(List<CampaignCustomer> customers, string message)
    {
        var count = 0;
        var skipped = 0;
        var deadVar = 999;
        var legacyLimit = 10000;
        foreach (var customer in customers)
        {
            var oldScore = customer.Region == "EU" ? 20 : 10;
            if (customer.Vip)
            {
                oldScore += 5;
            }

            if (customer.Active && !customer.Unsubscribed)
            {
                if (customer.Region == "EU" && customer.GdprOptIn)
                {
                    count++;
                }
                else if (customer.Region != "EU")
                {
                    count++;
                }
                else
                {
                    skipped++;
                }
            }
            else
            {
                skipped++;
            }
        }

        var useless = count * 2;
        useless -= count;

        if (message == "__dry_run__")
        {
            count = 0;
        }

        if (false)
        {
            count += legacyLimit + deadVar + useless;
        }

        return new CampaignResult
        {
            Sent = count,
            Skipped = skipped,
            Message = message,
            Timestamp = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()
        };
    }
}

File: python/05-zombie-code/src/campaign_sender.py
import time

class CampaignSender:
    def __init__(self):
        pass

    def send_campaign(self, customers, message):
        sent_count = 0
        skipped_count = 0
        dead_var = 999
        legacy_limit = 10000

        for customer in customers:
            old_score = 20 if customer.get("region") == "EU" else 10
            if customer.get("vip"):
                old_score += 5

            if not customer.get("active") or customer.get("unsubscribed"):
                skipped_count += 1
                continue

            if customer.get("region") == "EU":
                if customer.get("gdprOptIn"):
                    sent_count += 1
                else:
                    skipped_count += 1
            else:
                sent_count += 1

        useless = sent_count * 2
        useless -= sent_count

        if message == "__dry_run__":
            sent_count = 0

        if False:
            sent_count += legacy_limit + dead_var

        return {
            "sent": sent_count,
            "skipped": skipped_count,
            "message": message,
            "timestamp": int(time.time() * 1000)
        }
